### ISSUE TITLE
Feature: Visual indicator for grouped applications

### ISSUE THREAD

**Issue by marcosilva85** [dangvd/crystal-dock#34]
Is it possible to add a special indicator to identify grouped applications? It would enable us users to understand:
- we have multiple instances of same application opened
- which instance is selected at the moment

Perhaps a little number in the bottom right corner , or small circle with changing color as you switch the focus between app instances... or something else better of course, I'm just laying out some ideas.

**Comment by dangvd** [comment]
Hi,

Sure I'll experiment with this, probably for v2.4 or v2.5
For the next version v2.3 my priority is to bring back support for multi-screen #19

**Comment by marcosilva85** [comment]
cool @dangvd , in the event you'd like to see some ideas, here's how it looks on the custom widget [FancyTasks](https://github.com/alexankitty/FancyTasks) for KDE:
![ft](https://github.com/alexankitty/FancyTasks/raw/main/docs/metro.png?raw=true)

**Comment by dangvd** [comment]
Hi,

Thanks for the reference.

Btw here is what I'm currently experimenting with for Crystal Dock:
This show task count for each application as 1, 2 or 3+ (3+ only show as 3, due to lack of spaces)
The yellow indicator is the active task.

[
![task_count_indicator](https://github.com/user-attachments/assets/6b27b29d-113b-4703-a635-f974c88477af)
](url)

Need more work though.

**Comment by dangvd** [comment]
@marcosilva85 : so what do you think about the above prototype?

**Comment by marcosilva85** [comment]
I believe its an excelent way as well! It conveys the information!

**Comment by dangvd** [comment]
Thank you! I will get it in then, although I might turn it off by default because I think the current look (without the window count indicator) is cleaner and easier to the eyes. Then if you want to turn it on, you can open Task Manager Settings and change it there. How about that?

**Comment by dangvd** [comment]
Actually I might set it to default if I can find some good default colours.

**Comment by marcosilva85** [comment]
hi @dangvd, some comments from my side that might help,
> ... although I might turn it off by default because I think the current look (without the window count indicator) is cleaner and easier to the eyes

> Actually I might set it to default if I can find some good default colours.

I think your view on how that extra color would "feel off" its right, by nature of being an extra color on the eyes, but somehow I feel that picking a better color scheme would not "fix" it.

In app design when we need to convey info by adding extra colors on many things like text, buttons,etc. it starts to be to much, or "not pretty anymore".

However I think for this particular case one extra color in such a small "quantity" is not capturing to much eye-gaze and OK for the purpose.
Another way to encode the same meaning would be with a bar-and-dots approach - FancyTasks seems to have good ideas here.
A 3rd way: you could keep CrystalDock bars as they are, but keeping one hollow (the bar is just the outline). Or reverse, several hollow and one is filled, that might work the best. And, in the end, only one color ;)

**Comment by dangvd** [comment]
@marcosilva85 I already tried the bar-and-dots style but I found it not fitting for Crystal Dock's crystal/glass/3D look.

Anwyay I found better colours for my approach -- main thing is to increase contrast with other parts of the dock to make the smaller indicators easier to read. See the image below.

![task_count_indicator_2](https://github.com/user-attachments/assets/9620461b-aebb-4b9f-af6f-8c1031b3dc60)
